### DISCUSSION TITLE
Ignore inflections

### DISCUSSION THREAD

**Discussion by mortii** [mortii/anki-morphs#141] [Ideas]
> I still find the fact that I get so many variations of words I know. I have set it so I get 5 new cards per day the amount of cards I got with morphs that I already know is 30(marked as known). I had to see 6 times more cards than I would prefer

_Originally posted by @Vilhelm-Ian in https://github.com/mortii/anki-morphs/discussions/115#discussioncomment-8272951_

**Comment by mortii** [comment]
what are your U and A numbers @Vilhelm-Ian

**Reply by Vilhelm-Ian** [reply]
U: 3903, A: 11040

**Reply by mortii** [reply]
@Vilhelm-Ian wow, that is a big discrepancy.

I'm curious, when you use the readability report generator, what percentages do you normally get for german texts?

**Reply by Vilhelm-Ian** [reply]
I get 60% which is absurd. Since I can understand most books that are YA

**Reply by mortii** [reply]
I have U: 2740, A: 3091 for japanese, and the median readability report percent is ~85, standard deviation is ~5.

This lemma-inflection disparity is definitely a much bigger problem for german.

Did the last readability report generator update make the percentage any better than it was previously?

**Reply by Vilhelm-Ian** [reply]
yes I had 10% and now I have 60%. But realistically I should have somewehere above 90%

**Reply by mortii** [reply]
@Vilhelm-Ian could you [export your known morphs](https://mortii.github.io/anki-morphs/user_guide/usage/known-morphs-exporter.html) and upload the csv file here? It would help me get a better picture of the situation.

**Comment by Vilhelm-Ian** [comment]
[known_morphs-2024-01-31@09-45-33.csv](https://github.com/mortii/anki-morphs/files/14109312/known_morphs-2024-01-31%4009-45-33.csv)
@mortii

**Comment by mortii** [comment]
I think having an option in the readability report generator to ignore inflections can be good, as it could give a better estimate of what you might vaguely known, on top of your concrete knowledge.

I think you can try this out by simply removing the `+ morph.inflection`  part from `key` here:
https://github.com/mortii/anki-morphs/blob/2c2c8d45582dd5b20ad51282d8dc740a97c8d524/ankimorphs/readability_report_generator.py#L193-L205

It might require more tweaking, but that would be the general idea i think.

@Vilhelm-Ian do you want to give it a go and see if the results are better?

**Reply by Vilhelm-Ian** [reply]
this is really weird after deleting that part of the line. The known percentage dropped by 1%

**Reply by mortii** [reply]
@Vilhelm-Ian that is actually a very interesting result. You should create a short txt file and check if it gives accurate results for it or not, and then gradually increase the text in it to see if it holds true all the way up in length/complexity.

**Reply by Vilhelm-Ian** [reply]
will try

**Comment by mortii** [comment]
@Vilhelm-Ian I think I have this working in the ["known-lemma"](https://github.com/mortii/anki-morphs/tree/known-lemma) branch now (very ugly code currently). Could you test it and make sure? You have to select "lemma" as the morph priority in the "algorithm" tab in the settings. I'll move this option somewhere else later.

**Comment by mortii** [comment]
Included in the v3 update: https://github.com/mortii/anki-morphs/releases/tag/v3.0.0
